starfish/test/full_pipelines/cli/test_dartfish_cli.py

```diff
@@ -0,0 +1,104 @@
+import os
+import sys
+
+import numpy as np
+import pandas as pd
+
+from starfish import IntensityTable
+from starfish.test.full_pipelines.cli._base_cli_test import CLITest
+from starfish.types import Features
+
+
+class TestWithDartfishData(CLITest):
+    __test__ = True
+
+    SUBDIRS = (
+        "registered",
+        "filtered",
+        "results"
+    )
+
+    STAGES = (
+        [
+            sys.executable,
+            "examples/get_cli_test_data.py",
+            "https://dmf0bdeheu4zf.cloudfront.net/20180828/DARTFISH-TEST/dartfish-test-data.zip",
+            lambda tempdir, *args, **kwargs: os.path.join(tempdir, "registered")
+        ],
+        [
+            "starfish", "filter",
+            "--input", lambda tempdir, *args, **kwargs: os.path.join(
+                tempdir, "registered/fov_001", "hybridization.json"),
+            "--output", lambda tempdir, *args, **kwargs: os.path.join(
+                tempdir, "filtered", "scale_filtered.json"),
+            "ScaleByPercentile",
+            "--p", "100",
+        ],
+        [
+            "starfish", "filter",
+            "--input", lambda tempdir, *args, **kwargs: os.path.join(
+                tempdir, "filtered", "scale_filtered.json"),
+            "--output", lambda tempdir, *args, **kwargs: os.path.join(
+                tempdir, "filtered", "zero_filtered.json"),
+            "ZeroByChannelMagnitude",
+            "--thresh", ".05",
+        ],
+        [
+            "starfish", "detect_spots",
+            "--input", lambda tempdir, *args, **kwargs: os.path.join(
+                tempdir, "filtered", "zero_filtered.json"),
+            "--output", lambda tempdir, *args, **kwargs: os.path.join(tempdir, "results"),
+            "PixelSpotDetector",
+            "--codebook", lambda tempdir, *args, **kwargs: os.path.join(
+                tempdir, "registered", "codebook.json"),
+            "--distance-threshold", "3",
+            "--magnitude-threshold", ".5",
+            "--min-area", "5",
+            "--max-area", "30",
+        ],
+    )
+
+    def verify_results(self, intensities):
```

Comment:
This makes me think we should refactor the API tests to generate an object that this can check against as well. That way we have one source of truth. Maybe you're already ahead of me? :-)